BootloaderCorePkg/Tools/GenCapsuleFirmware.py

```diff
@@ -15,105 +15,532 @@
+        except:
+            Message = '{Argument} is not a valid registry format GUID value.'.format(
+                Argument=Argument)
+            raise argparse.ArgumentTypeError(Message)
+        return Value
+
+    #
+    # Create command line argument parser object
+    #
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        '-p',
+        '--payload',
+        nargs=2,
+        action='append',
+        type=str,
+        required=True,
+        help=
+        'Specify payload information including GUID, FileName')
+    parser.add_argument(
+        '-k',
+        '--priv_key',
+        dest='PrivKey',
+        type=str,
+        required=True,
+        help='Private RSA 2048 key in PEM format to sign image')
+    parser.add_argument('-o',
+                        '--output',
+                        dest='NewImage',
+                        type=str,
+                        required=True,
+                        help='Output file for signed image')
+
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        dest='Verbose',
+        action="store_true",
+        help=
+        "Turn on verbose output with informational messages printed, including capsule headers and warning messages.")
+
+    #
+    # Parse command line arguments
+    #
+    args = parser.parse_args()
+
+    FmpCapsuleHeader = FmpCapsuleHeaderClass()
+    for PldUuid, PldFile in args.payload:
+        
+        FmpPayloadHeader = FmpPayloadHeaderClass()
+        Buffer = open(PldFile, 'rb').read()
+        Result = Buffer
+        FmpPayloadHeader.Payload = Result
+        Result = FmpPayloadHeader.Encode()
+
+        if PldUuid in PredefinedUuidDict:
+            PldUuid = PredefinedUuidDict[PldUuid]
+        Guid = ValidateRegistryFormatGuid(PldUuid)
+        FmpCapsuleHeader.AddPayload(Guid, Result, HardwareInstance=0)
+
+    Result = FmpCapsuleHeader.Encode()
+
+    #
+    # Write binary output file
+    #
+    if args.Verbose:
+        FmpCapsuleHeader.DumpInfo()
+
+    #
+    # Create final capsule
+    #
+    SignImage(Result, args.NewImage, args.PrivKey)
+
+    print('Success')
 
-  if os.name == 'nt':
-    os.environ['OPENSSL_PATH'] = "C:\\Openssl\\"
-
-  #
-  # Create command line argument parser object
-  #
-  parser = argparse.ArgumentParser()
-  parser.add_argument('-b',  '--image', dest='Image', type=str, help='Raw image into firmware update capsule image', required = True)
-  parser.add_argument('-t',  '--type', dest='Type', choices=['b','c'], help='Image type b:Bootloader c:CfgData', default = 'b')
-  parser.add_argument('-k',  '--priv_key', dest='PrivKey', type=str, help='Private RSA 2048 key in PEM format to sign image', required = True)
-  parser.add_argument('-o',  '--output', dest='NewImage', type=str, help='Output file for signed image', required = True)
-  parser.add_argument("-q", "--quiet", dest='Quiet', action="store_true", help="without output messages or temp files")
```

Comment:
If removed the arguments, how it could support the parameter "-p", "-o" and "-k" in your commit message?

Reply:
If you see what are removed, these are all unused args, Look from line# 513 in the new file, args are parsed there

Reply:
sorry I missed the one you added since code added/removed are not back to back in the patch.
That's why we prefer only changing necessary code in patch.  The new format is not aligned in different parameter. the old one is more readable.